src/ucp/wireup/select.c

```diff
@@ -1482,9 +1482,17 @@ ucp_wireup_construct_lanes(const ucp_wireup_select_params_t *select_params,
         addr_indices[lane]            = select_ctx->lane_descs[lane].addr_index;
 
         if (select_ctx->lane_descs[lane].usage & UCP_WIREUP_LANE_USAGE_CM) {
-            ucs_assert((key->cm_lane == UCP_NULL_LANE) ||
-                       (key->cm_lane == lane));
-            key->cm_lane = lane;
+            if (select_params->ep_init_flags & UCP_EP_INIT_CM_WIREUP_CLIENT) {
+                /* Client creates CM lane before construction TL lanes */
+                ucs_assert((lane == 0) && (key->cm_lane == 0));
```

Comment:
not sure we need this. do we assume cm_lane is 0 anywhere in the code?

Reply:
No, only for client because cm_lane is a start point of wireup, for server side it's just because its added to begin of ucp_wireup_search_lanes. BTW, might be better to move it to the end and assume that cm_lane is last on server side to make loop with "connect lane" logic simpler.